File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        lpp = 0
        rpp = len(numbers) - 1
        total = numbers[lpp] + numbers[rpp]

        while total != target:
            if total > target:
                rpp -=1
            if total < target:
                lpp += 1
            total = numbers[lpp] + numbers[rpp]

        return [lpp+1, rpp+1 ]
        

        

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-0.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {

        int result = nums.length + 1;
        int total = 0;
        int l = 0;

        for (int r = 0; r < nums.length; r++){
            total += nums[r];
            while (total >= target){
                result = (Math.min(r-l, result));
                total -= nums[l];
                l++;
            }
        }

        if (result == nums.length + 1) return 0;
        return result + 1;
    }
}